Objects/funcobject.c

```diff
@@ -423,6 +423,29 @@ func_get_annotations(PyFunctionObject *op, void *Py_UNUSED(ignored))
         op->func_annotations = PyDict_New();
         if (op->func_annotations == NULL)
             return NULL;
+
+    }
+    if (PyTuple_CheckExact(op->func_annotations)) {
+        PyObject *anns = op->func_annotations;
+        op->func_annotations = PyDict_New();
```

Comment:
When some error happened in this `if` block, `op->func_annotations` can be incomplete dict.
Then, when `func.__annotations__` is accessed next time, the incomplete dict can be returned.

To avoid returning incomplete dict:

```c
PyObject *anns = PyDict_New();

// fill anns...

// finally
Py_SETREF(op->func_annotations, anns);
```

Reply:
Nice catch, you are totally right. And thank you for the solution.